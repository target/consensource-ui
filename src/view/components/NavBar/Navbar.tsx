import React, { FC } from 'react';
import {
  AppBar,
  Toolbar,
  Typography,
  makeStyles,
  Divider,
  Grid,
  createStyles,
} from '@material-ui/core';
import { ProfileMenu } from './ProfileMenu';
import { NavbarLink } from './NavbarLink';
import { UnstyledLink } from '../Links/UnstyledLink';

// Height of a "dense" toolbar
export const NAVBAR_HEIGHT = 48;

const useStyles = makeStyles(({ palette, zIndex }) =>
  createStyles({
    navbar: {
      zIndex: zIndex.drawer + 1,
    },
    logoContainer: {
      flexGrow: 1,
    },
    persona: {
      fontWeight: 'lighter',
    },
    divider: {
      backgroundColor: palette.primary.contrastText,
    },
    authItemsPadding: {
      paddingRight: 10,
    },
  }),
);

export const BaseNavBar: FC = ({ children: AuthItems }) => {
  const classes = useStyles();

  // TODO: Replace with SVG
  const Logo = () => (
    <div className={classes.logoContainer}>
      <UnstyledLink to="/">
        <Grid container spacing={1}>
          <Grid item>
            <Typography variant="h5">ConsenSource</Typography>
          </Grid>
          <Grid item>
            <Divider orientation="vertical" className={classes.divider} />
          </Grid>
          <Grid item>
            <Typography variant="h6" className={classes.persona}>
              Retailer
            </Typography>
          </Grid>
        </Grid>
      </UnstyledLink>
    </div>
  );

  return (
    <AppBar position="fixed" color="primary" className={classes.navbar}>
      <Toolbar variant="dense">
        {/* Note that all items after <Logo /> will be on the right side
        of the NavBar due to the `flexGrow: 1` styling */}
        <Logo />
        <NavbarLink label="About" route="http://consensource.io" openInNewTab />
        <NavbarLink label="Search Factories" route="/search" />
        {AuthItems}
      </Toolbar>
    </AppBar>
  );
};

export const UnauthedNavbar = () => {
  return (
    <BaseNavBar>
      <NavbarLink label="Login" route="/login" />
    </BaseNavBar>
  );
};

export const AuthedNavbar = () => {
  return (
    <BaseNavBar>
      <ProfileMenu />
    </BaseNavBar>
  );
};
