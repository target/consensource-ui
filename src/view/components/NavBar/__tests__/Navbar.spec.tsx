import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import { AuthedNavbar, UnauthedNavbar } from '../Navbar';

describe('<AuthedNavbar />', () => {
  it('renders', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/']}>
        <Route path="/">
          <AuthedNavbar />
        </Route>
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});

describe('<UnauthedNavbar />', () => {
  it('renders', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/']}>
        <Route path="/">
          <UnauthedNavbar />
        </Route>
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
