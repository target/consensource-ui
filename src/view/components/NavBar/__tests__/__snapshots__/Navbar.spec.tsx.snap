// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthedNavbar /> renders 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-navbar-1 mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
    >
      <div
        class="makeStyles-logoContainer-2"
      >
        <a
          class="makeStyles-link-6"
          href="/"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5"
              >
                ConsenSource
              </h5>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <hr
                class="MuiDivider-root makeStyles-divider-4 MuiDivider-vertical"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <h6
                class="MuiTypography-root makeStyles-persona-3 MuiTypography-h6"
              >
                Retailer
              </h6>
            </div>
          </div>
        </a>
      </div>
      <a
        class="makeStyles-link-6"
        href="http://consensource.io"
        target="_blank"
      >
        <p
          class="MuiTypography-root makeStyles-link-7 false MuiTypography-body1"
        >
          About
        </p>
      </a>
      <a
        class="makeStyles-link-6"
        href="/search"
        target=""
      >
        <p
          class="MuiTypography-root makeStyles-link-7 false MuiTypography-body1"
        >
          Search Factories
        </p>
      </a>
      <div>
        <button
          aria-label="profile"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </div>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
</div>
`;

exports[`<UnauthedNavbar /> renders 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-navbar-9 mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
    >
      <div
        class="makeStyles-logoContainer-10"
      >
        <a
          class="makeStyles-link-14"
          href="/"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5"
              >
                ConsenSource
              </h5>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <hr
                class="MuiDivider-root makeStyles-divider-12 MuiDivider-vertical"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <h6
                class="MuiTypography-root makeStyles-persona-11 MuiTypography-h6"
              >
                Retailer
              </h6>
            </div>
          </div>
        </a>
      </div>
      <a
        class="makeStyles-link-14"
        href="http://consensource.io"
        target="_blank"
      >
        <p
          class="MuiTypography-root makeStyles-link-15 false MuiTypography-body1"
        >
          About
        </p>
      </a>
      <a
        class="makeStyles-link-14"
        href="/search"
        target=""
      >
        <p
          class="MuiTypography-root makeStyles-link-15 false MuiTypography-body1"
        >
          Search Factories
        </p>
      </a>
      <a
        class="makeStyles-link-14"
        href="/login"
        target=""
      >
        <p
          class="MuiTypography-root makeStyles-link-15 false MuiTypography-body1"
        >
          Login
        </p>
      </a>
    </div>
  </header>
</div>
`;
